### PR TITLE
Experience Fragments Smoke Tests

### PR DESCRIPTION
**Pull request by stefangrimm**
- Added tests for creating and deleting variations
- Cleaned up Content Package
- Created one single test class for Create/Delete XF/variation to improve performance

Note: I've kept the different steps to be able to revert them if necessary. Easiest way to review is to check the end result.

### CODE REVIEW

File: xf-smoke/src/main/java/com/adobe/cq/cloud/testing/it/xf/smoke/Constants.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.adobe.cq.cloud.testing.it.xf.smoke;
+
+final class Constants {
```
Comment (volteanu) [inline xf-smoke/src/main/java/com/adobe/cq/cloud/testing/it/xf/smoke/Constants.java]: I think this class doesn't bring much value, each test class can define its own values and it's just a coincidence the values are the same.

**Review by volteanu — APPROVED** [review]
LGTM overall!

File: xf-smoke/src/main/java/com/adobe/cq/cloud/testing/it/xf/smoke/Constants.java (hunk 1 above)
Comment (stefangrimm) [reply xf-smoke/src/main/java/com/adobe/cq/cloud/testing/it/xf/smoke/Constants.java]: I've removed the `Constants` class. It turned out that the content package is not even required for `XfSmokeIT`, so I'm fine with the suggested change (no duplication in this case).

**Comment by stefangrimm** [comment]
I have also reviewed the content package once more and changed the following:

- Removed the `we-retail` labelled content, as it is not required by these tests.
- Added a suffix to the filter roots to make the surface for clashes with customer content as small as possible.

Note that if we would get rid of the `ConvertToXfIT`, we could get rid of the content package as well. IMHO `ConvertToXfIT` is not a typical smoke test.

WDYT?

Note that I didn't set up a Jenkins task, but have run the tests multiple times against a Skyline instance and saw no flakiness. Also, the original (= unadapted) tests are regularly run as part of our internal "Sanity test" Jenkins jobs. If that's not good enough, I guess @tyge68 could set up such a task.

As I am on PTO for the next ~4 weeks, I'll have to hand this over to @martinischeery now, for dispatching it further. Might be necessary to create another fork, unfortunately.

**Comment by martinischeery** [comment]
@volteanu Could you please merge this one?

File: xf-smoke/src/main/resources/test-content/jcr_root/content/.content.xml [hunk 2]
```diff
@@ -26,5 +26,5 @@
     <geometrixx/>
```
Comment (dulvac) [inline xf-smoke/src/main/resources/test-content/jcr_root/content/.content.xml]: @stefangrimm  Can we remove all these folders? geometrixx, phonegap, social, we-retail etc? They don't make any sense on skyline and I'm not sure why they were there in the first place.

**Review by dulvac — COMMENTED** [review]
@stefangrimm , I see some references to some unneeded (?) folders like geometrixx

**Comment by dulvac** [comment]
@martinischeery , @tyge68 , if we can get rid of the content package by removing the convert xf it, I would think that makes sense! I agree with @stefangrimm it's not really a smoke test.

@martinischeery , can someone from your team take care of that interruptively?

**Comment by dobretony** [comment]
As per recommendation I've removed ConvertToXfIT and as such managed to get rid of all the content.
I've verified on the FluidXP skyline instances and 6 tests are run without issue ( Create XF, Create XF in Folder, Delete XF, Create Variation, Delete Variation and Delete Master Variation ).

@dulvac @volteanu Could you take a look again?

**Review by volteanu — APPROVED** [review]
Nice!

**Comment by volteanu** [comment]
thanks @stefangrimm and @dobretony ! Your changes were released in: https://github.com/adobe/aem-test-samples/tree/com.adobe.cq.cloud.testing.it.xf.smoke-0.2.0

**Pull request merged**